viewing_party/party.py

```diff
@@ -1,23 +1,231 @@
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+
+    #return None of title, genre, and rating are empty
+    if not (title and genre and rating):
+        return None
+    
+    movie = {
+                "title": title,
+                "genre": genre,
+                "rating": rating
+            
+            }
+    return movie
+
+
+def add_to_watched(user_data, movie):
+
+    user_data["watched"].append(movie)
+    
+    return user_data
+
+
+def add_to_watchlist(user_data, movie):
+
+    user_data["watchlist"].append(movie)
+    
+    return user_data
+
+
+def watch_movie(user_data, title): 
+
+    #loop through each movie in the watchlist
+    for movie in user_data["watchlist"]:
+
+        #if it matches the title passed as the arg add it to watch
+        if title == movie["title"]:
+
+            user_data["watched"].append(movie)
+            user_data["watchlist"].remove(movie)

```

Comment:
We want to avoid the anti-pattern of removing a value from a list we are actively iterating over. After the first removal, we skip a value because the way iterators work (recall our discussion on the "danger zone" from the roundtable on "Iterating over Data"). Given the data for our tests, we know this isn't a concern for this application, but what if we were working with data that had duplicate titles?

Another consideration: From a performance/time standpoint, consider if we were dealing with a list of 1,000 movies. If the item we wanted to swap happens to be the first item, we would continue looping over 999 titles before the loop exits. Exiting early also helps us avoid extra time and calculations that don't add to our final result.